timemachine/md/smc.py

```diff
@@ -0,0 +1,179 @@
+# adapted from https://github.com/proteneer/timemachine/blob/7d6099b0f5b4a2d0b26c3edc7a91c18f7a526c00/md/experimental/smc.py
+
+from typing import Any, Callable, Dict, List, Tuple
+
+import numpy as np
+from scipy.special import logsumexp
+from typing_extensions import TypeAlias
+
+# type annotations
+Sample: TypeAlias = Any
+Samples: TypeAlias = List[Sample]  # e.g. List[CoordsVelBox]
+
+Lambda: TypeAlias = float
+LogWeight: TypeAlias = float
+Array: TypeAlias = np.ndarray
+IndexArray: TypeAlias = Array
+LogWeights: TypeAlias = Array
+
+BatchPropagator = Callable[[Samples, Lambda], Samples]
```

Comment:
nit: why use `TypeAlias` annotations for some aliases and not others?

Reply:
Only added where needed for mypy. The aliases need this annotation, but the `Callable` type declaration does not, I think

Reply:
Oh! I was confused about whether this was necessary. To fix the mypy error I had done two things at once: put declarations on separate lines, and annotate with :TypeAlias https://github.com/proteneer/timemachine/pull/718/commits/78e8edabfba371c27b41d305822895d9548e9bcf . However, it looks like all of the :TypeAlias annotations may have been unnecessary, and I've now removed them in: https://github.com/proteneer/timemachine/pull/718/commits/5baafa71d9e482035fc7388da3d6063d741615fd